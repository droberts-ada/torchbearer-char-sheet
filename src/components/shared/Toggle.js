import React from 'react';
import PropTypes from 'prop-types';

import Checkbox from './Checkbox';

// TODO DPR: come up with a more generic way to pass in the test
const Toggle = ({ name, active, test, subtext, onToggle, disabled }) => {
  console.log(name);
  console.log(subtext);
  let className = "toggle";
  if (active) {
    className += " active";
  }
  if (!test) {
    className += " notest";
  }
  return (
    <li className={className}>
      <div>
        <h3>{ name }</h3>
        { test ? (<p>(Ob { test.ob } { test.skill })</p>) : ""}
        <Checkbox onToggle={onToggle} active={active} disabled={disabled} />
      </div>
      { subtext ? (<p>{ subtext }</p>) : "" }
    </li>
  );
};

Toggle.propTypes = {
  name: PropTypes.string.isRequired,
  active: PropTypes.bool.isRequired,
  subtext: PropTypes.string,
  onToggle: PropTypes.func.isRequired,
  // test: PropTypes.shape({ ob: PropTypes.number, skill: PropTypes.string }),
};

export default Toggle;
