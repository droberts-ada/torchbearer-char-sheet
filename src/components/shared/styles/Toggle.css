.toggle > div {
  display: grid;
  grid-template-columns: 1fr 1fr 2em;
  align-items: end;
}

.toggle.notest h3 {
  grid-column: 1 / 3;
}

.toggle .checkbox {
  grid-column: 3;
}
