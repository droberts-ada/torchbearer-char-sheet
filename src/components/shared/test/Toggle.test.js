import React from 'react';
import renderer from 'react-test-renderer';

import Toggle from '../Toggle';

it('renders when active', () => {
  const tree = renderer
  .create(
    <Toggle
      name="Test toggle"
      subtext="some subtext for a toggle"
      active={true}
      onToggle={jest.fn()}
      />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders when inactive', () => {
  const tree = renderer
  .create(
    <Toggle
      name="Test toggle"
      subtext="some subtext for a toggle"
      active={false}
      onToggle={jest.fn()}
      />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders when disabled', () => {
  const tree = renderer
  .create(
    <Toggle
      name="Test toggle"
      subtext="some subtext for a toggle"
      active={false}
      disabled={true}
      onToggle={jest.fn()}
      />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
