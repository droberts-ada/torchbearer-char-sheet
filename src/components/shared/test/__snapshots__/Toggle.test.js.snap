// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when active 1`] = `
<li
  className="control toggle active notest"
>
  <div>
    <h3>
      Test toggle
    </h3>
    
    <div
      className="checkbox"
      onClick={[Function]}
      style={
        Object {
          "backgroundImage": "url(letter-x.png)",
        }
      }
    />
  </div>
  <p>
    some subtext for a toggle
  </p>
</li>
`;

exports[`renders when disabled 1`] = `
<li
  className="control toggle notest"
>
  <div>
    <h3>
      Test toggle
    </h3>
    
    <div
      className="checkbox disabled"
      onClick={[Function]}
      style={undefined}
    />
  </div>
  <p>
    some subtext for a toggle
  </p>
</li>
`;

exports[`renders when inactive 1`] = `
<li
  className="control toggle notest"
>
  <div>
    <h3>
      Test toggle
    </h3>
    
    <div
      className="checkbox"
      onClick={[Function]}
      style={undefined}
    />
  </div>
  <p>
    some subtext for a toggle
  </p>
</li>
`;
